plugins/modules/monitoring/pagerduty_user.py

```diff
@@ -0,0 +1,263 @@
+                self._module.fail_json(
+                    msg="Failed to remove %s as user was not found" % (pd_name))
+            if e.response.status_code == 403:
+                self._module.fail_json(
+                    msg="Failed to remove %s due to inability to review the requested resource within the API token" % (pd_name))
+            if e.response.status_code == 401:
+                # print out the list of incidents
+                pd_incidents = self.get_incidents_assigned_to_user(pd_user_id)
+                self._module.fail_json(msg="Failed to remove %s as user has assigned incidents %s" % (pd_name, pd_incidents))
+            if e.response.status_code == 429:
+                self._module.fail_json(
+                    msg="Failed to remove %s due to reaching the limit of making requests" % (pd_name))
+
+    # get incidents assigned to a user
+    def get_incidents_assigned_to_user(self, pd_user_id):
+        incident_info = {}
+        incidents = self._apisession.list_all('incidents', params={'user_ids[]': [pd_user_id]})
+
+        for incident in incidents:
+            incident_info = {
+                'title': incident['title'],
+                'key': incident['incident_key'],
+                'status': incident['status']
+            }
+        return incident_info
+
+    # add a user to a team/teams
+    def add_user_to_teams(self, pd_user_id, pd_teams, pd_role):
+        updated_team = None
+        for team in pd_teams:
+            team_info = self._apisession.find('teams', team, attribute='name')
+            if team_info is not None:
+                try:
+                    updated_team = self._apisession.rput('/teams/' + team_info['id'] + '/users/' + pd_user_id, json={
+                        'role': pd_role
+                    })
+                except PDClientError:
+                    updated_team = None
+        return updated_team
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            access_token=dict(type='str', required=True, no_log=True),
+            pd_user=dict(type='str', required=True),
+            pd_email=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            pd_role=dict(type='str', default='responder',
+                         choices=['global_admin', 'manager', 'responder', 'observer', 'stakeholder', 'limited_stakeholder', 'restricted_access']),
+            pd_teams=dict(type='list', elements='str', required=False)),
+        required_if=[['state', 'present', ['pd_teams']], ],
+        supports_check_mode=True,
+    )
+
+    if not HAS_PD_PY:
+        module.fail_json(msg=missing_required_lib('pdpyras', url='https://github.com/PagerDuty/pdpyras'), exception=PD_IMPORT_ERR)
+
+    if not HAS_PD_CLIENT_ERR:
+        module.fail_json(msg=missing_required_lib('PDClientError', url='https://github.com/PagerDuty/pdpyras'), exception=PD_CLIENT_ERR_IMPORT_ERR)
```

Comment:
Is this a separate library that can be installed with `pip install PDClientError`, or is this a part of `pdpyras` that's always there?

Reply:
It is a part of `pdpyras`.

Reply:
In that case, one of the try/except suffices (and one error handling). Simply put both imports into the same try/except.